packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -266,6 +266,7 @@ abstract class CachingAssetBundle extends AssetBundle {
       .then<T>(parser)
       .then<void>((T value) {
         result = SynchronousFuture<T>(value);
+        _structuredBinaryDataCache[key] = result!;
```

Comment:
How about a unit test that directly invokes loadStructuredBinaryData and asserts that the cache is populated sychonously?